### PR TITLE
fix: view post/page from WP admin

### PR DESCRIPTION
**Pull request by royelkhoury**
## Description:
The purpose of this PR is to enable view posts and pages from WP admin post/page list.
Refer to [this issue](https://github.com/postlight/headless-wp-starter/issues/186)

#### Before
Screen Recording 2022-02-07 at 4 41 10 PM

#### After
Screen Recording 2022-02-07 at 4 39 30 PM

### CODE REVIEW

File: frontend/pages/preview.js [hunk 1]
```diff
@@ -7,6 +7,12 @@ import Menu from '../components/Menu';
 import Config from '../config';
 
 class Preview extends Component {
+  static async getInitialProps(context) {
+    const { id, rev, type, status, wpnonce } = context.query;
+    let url = { query: { id, rev, type, status, wpnonce } };
```
Comment (alexiakl) [inline frontend/pages/preview.js:12]: url looks to be immutable, it should be a const

File: frontend/pages/preview.js [hunk 2]
```diff
@@ -22,16 +28,16 @@ class Preview extends Component {
 
     // checking if the post/page is a draft or a revision.
     let postUrl = `${Config.apiUrl}/wp/v2/${type}s/${id}/revisions/${rev}?_wpnonce=${wpnonce}`;
-    if( status === 'draft' ) {
+    if (status === 'draft') {
```
Comment (alexiakl) [inline frontend/pages/preview.js:31]: spacing changes here needs to be reverted, they are introduced by a php linter

File: frontend/pages/preview.js [hunk 3]
```diff
@@ -22,16 +28,16 @@ class Preview extends Component {
...
       postUrl = `${Config.apiUrl}/wp/v2/${type}s/${rev}?_wpnonce=${wpnonce}`;
     }
 
     fetch(
       postUrl,
       { credentials: 'include' }, // required for cookie nonce auth
     )
-      .then(res => res.json())
-      .then(res => {
+      .then((res) => res.json())
```
Comment (alexiakl) [inline frontend/pages/preview.js:39]: same here. revert the parentheses changes please

File: wordpress/wp-content/themes/postlight-headless-wp/inc/admin.php [hunk 4]
```diff
@@ -46,6 +46,9 @@ function set_headless_preview_link( $link ) {
 
 add_filter( 'preview_post_link', 'set_headless_preview_link' );
 
+add_filter( 'post_link', 'set_headless_preview_link' );
```
Comment (alexiakl) [inline wordpress/wp-content/themes/postlight-headless-wp/inc/admin.php:49]: I need to check this closer - so I'll run the project, but it looks like you're setting the link of normal posts and pages as a preview link. No action needed here for now, but I gonna check few things

**Review by alexiakl — CHANGES_REQUESTED** [review]
Thanks for picking this up Roy.
I requested few changes

**Comment by royelkhoury** [comment]
Thanks @alexiakl for the feedback, changes found in 4fac37e

File: frontend/pages/preview.js [hunk 5]
```diff
@@ -7,10 +7,8 @@ import Menu from '../components/Menu';
 import Config from '../config';
 
 class Preview extends Component {
-  static async getInitialProps(context) {
-    const { id, rev, type, status, wpnonce } = context.query;
-    const url = { query: { id, rev, type, status, wpnonce } };
-    return { url };
+  static async getInitialProps({ query }) {
+    return { url: { query } };
```
Comment (reemobeid17) [inline frontend/pages/preview.js:11]: Have a look at the refactored code @royelkhoury @alexiakl

**Review by alexiakl — APPROVED** [review]

**Pull request merged**